flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql/SqlModelCall.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.calcite.sql;
+
+import org.apache.flink.table.planner.catalog.CatalogSchemaModel;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.sql.validate.SqlValidatorScope;
+
+import static org.apache.calcite.util.Static.RESOURCE;
+
+/** SqlModelCall to fetch and reference model based on identifier. */
+public class SqlModelCall extends SqlBasicCall {
+
+    private final CatalogSchemaModel model;
+
+    public SqlModelCall(SqlExplicitModelCall modelCall, CatalogSchemaModel model) {
+        super(
+                new SqlModelOperator(model),
+                modelCall.getOperandList(),
+                modelCall.getParserPosition(),
+                modelCall.getFunctionQuantifier());
+        this.model = model;
+    }
+
+    @Override
+    public void validate(SqlValidator validator, SqlValidatorScope scope) {
+        SqlIdentifier modelIdentifier = (SqlIdentifier) getOperandList().get(0);
+        if (model == null) {
```

Comment:
Remove this check. It is useless now.

Reply:
We need to override this and avoid validating identifier since it will be treated as table column. So I just put a check here since we didn't check null in constructor

Reply:
I think we can override this method, but we don't need to check model is null or not.